openpype/hosts/maya/api/pipeline.py

```diff
@@ -681,6 +685,105 @@ def before_workfile_save(event):
         create_workspace_mel(workdir_path, project_name)
 
 
+def display_warning(message, show_cancel=False):
+    """Show feedback to user.
+
+        Returns:
+            bool
+        """
+
+    from qtpy import QtWidgets
+
+    accept = QtWidgets.QMessageBox.Ok
+    if show_cancel:
+        buttons = accept | QtWidgets.QMessageBox.Cancel
+    else:
+        buttons = accept
+
+    state = QtWidgets.QMessageBox.warning(
+        None,
+        "",
+        message,
+        buttons=buttons,
+        defaultButton=accept
+    )
+
+    return state == accept
+
+
+def workfile_save_before_xgen(event):
+    current_work_dir = legacy_io.Session["AVALON_WORKDIR"].replace("\\", "/")
+    expected_work_dir = event.data["workdir_path"].replace("\\", "/")
+    if current_work_dir == expected_work_dir:
+        return
+
+    palettes = cmds.ls(type="xgmPalette", long=True)
+    if not palettes:
+        return
+
+    import xgenm
```

Comment:
Can we add a comment as to why we're importing no earlier than here - is it due to slowness? or required plugin load?

Reply:
For ensuring the module is available, so will move to just below the plugin load check.